kf_lib_data_ingest/validation/reporting/base.py

```diff
@@ -0,0 +1,202 @@
+"""
+Abstract base class for validation report builders
+"""
+import os
+import logging
+from abc import ABC, abstractmethod
+from pprint import pformat
+
+from kf_lib_data_ingest.common.type_safety import (
+    assert_safe_type,
+    assert_all_safe_type,
+)
+
+SAMPLE_VALIDATION_RESULTS = os.path.abspath("./sample_validation_results.py")
+REQUIRED_RESULT_KEYS = {"type", "description", "is_applicable", "errors"}
+PASSED = "success"
+FAILED = "fail"
+NA = "did not run"
+RESULT_TO_EMOJI = {PASSED: "✅", FAILED: "❌", NA: "🔘"}
+
+
+class AbstractReportBuilder(ABC):
+    def __init__(self, output_dir=None, setup_logger=False):
```

Comment:
Considering how large this class is, maybe it'd help to have at least a short blurb on what it's used for. It being an abc gives me some idea but I'm not totally sure without going through line by line 😁